backend/src/pong-game/entities/pong-game.entity.ts

```diff
@@ -0,0 +1,218 @@
+import { ObjectType, Field, Int, InputType } from '@nestjs/graphql'
+
+@ObjectType()
+class playfield {
+  @Field(() => Int)
+  width = 800
+
+  @Field(() => Int)
+  height = 600
+}
+
+@ObjectType()
+class racket {
+  constructor(hPos: number) {
+    this.hPos = hPos
+  }
+
+  @Field(() => Int)
+  width = 10
+
+  @Field(() => Int)
+  height = 60
+
+  @Field(() => Number)
+  hPos: number
+
+  @Field(() => Number)
+  vPos = 270
+
+  @Field(() => Int)
+  velocity = 6
+}
+
+@ObjectType()
+class ball {
+  @Field(() => Int)
+  radius = 16
+
+  @Field(() => Number)
+  hPos = 392
+
+  @Field(() => Number)
+  vPos = 292
+
+  dir: { x: number; y: number } = { x: 100, y: 100 }
+}
+
+@ObjectType()
+export class Controls {
+  @Field(() => Boolean)
+  Z_Key = false
+
+  @Field(() => Boolean)
+  S_Key = false
+
+  @Field(() => Boolean)
+  Up_Key = false
+
+  @Field(() => Boolean)
+  Down_Key = false
+}
```

Comment:
Controls class present two times here and in player control input

Reply:
grpahql does not allox InputType() and ObjectType() for the same classe and for validation pipe.
Controls in game entity is used as part of the game class and control input is used to specified the controls send as argument for the resolver